wcc-contentful/lib/wcc/contentful/simple_client.rb

```diff
@@ -30,21 +30,24 @@ class SimpleClient
     # @param [String] space The Space ID to access
     # @param [String] access_token A Contentful Access Token to be sent in the Authorization header
     # @param [Hash] options The remaining optional parameters, defined below
-    # @option options [Symbol, Object] adapter The Adapter to use to make requests.
+    # @option options [Symbol, Object] connection The Faraday connection to use to make requests.
     #   Auto-discovered based on what gems are installed if this is not provided.
     # @option options [String] default_locale The locale query param to set by default.
     # @option options [String] environment The contentful environment to access. Defaults to 'master'.
     # @option options [Boolean] no_follow_redirects If true, do not follow 300 level redirects.
+    # @option options [Number] rate_limit_wait_timeout The maximum time to block the thread waiting
+    #   on a rate limit response.  By default will wait for one 429 and then fail on the second 429.
     def initialize(api_url:, space:, access_token:, **options)
       @api_url = URI.join(api_url, '/spaces/', space + '/')
       @space = space
       @access_token = access_token
 
-      @adapter = SimpleClient.load_adapter(options[:adapter])
+      @adapter = SimpleClient.load_adapter(options[:connection])
 
       @options = options
       @query_defaults = {}
       @query_defaults[:locale] = @options[:default_locale] if @options[:default_locale]
+      @rate_limit_wait_timeout = @options[:rate_limit_wait_timeout] || 1.5
```

Comment:
does this value come from contentful docs?

Reply:
The rate limit is 78 requests per second, then it asks you to wait 1 second.  This was my way of saying that we should wait long enough to retry once, but not twice.